### ISSUE TITLE
Guidance on how to use the AddHttpCorrelation extension method

### ISSUE THREAD

**Issue by fgheysels** [arcus-azure/arcus.webapi#284]
When using the `AddHttpCorrelation` extension method which takes no parameters, the compiler emits a warning saying that this method is deprecated, and the overload that takes `HttpCorrelationInfoOptions` must be used.

![image](https://user-images.githubusercontent.com/3605786/151523114-4b22ad3c-1251-4cda-af6d-2fd8b65976e3.png)

There exists an overload which takes `HttpCorrelationInfoOptions` instead of `CorrelationInfoOptions` but you need to use it like this;:

```csharp
Action<HttpCorrelationInfoOptions> action = options =>
{
   options.UpstreamService = ?
   options.Transaction = ?
   options.Operation = ?
};
services.AddHttpCorrelation(action);
```
But it is not clear to me how this can be used.   I suppose you just pass in the `TransactionId` and `OperationId` that must be used, but it would be good if we have an example of this.

Also, can we have a simple overload which falls back to default values ?

**Comment by stijnmoreels** [comment]
1. Yes, the first one is correct, you need to use (for now) typed lambda's to fix this, as the 'old' overload was using optional arguments and the new one uses method overloads. We hope to fix this in the next major.
2. The feature docs explains every option with examples: https://webapi.arcus-azure.net/features/correlation#configuration , not sure what you're referring to otherwise?

**Comment by fgheysels** [comment]
When we have [this](https://github.com/arcus-azure/arcus.webapi/issues/293) we can close this issue as well I think.

**Comment by stijnmoreels** [comment]
> When we have [this](https://github.com/arcus-azure/arcus.webapi/issues/293) we can close this issue as well I think.

Yes! Let's hope that change will clarify things.

**Comment by stijnmoreels** [comment]
Is fixed in #336 as the default method is now free again.
